rotkehlchen/exchanges/okx.py

```diff
@@ -0,0 +1,418 @@
+                self.msg_aggregator.add_error(
+                    f'Error processing {self.name} {asset.name} balance result due to inability '
+                    f'to deserialize asset amount due to {str(e)}. Skipping balance result.',
+                )
+                continue
+
+            assets_balance[asset] += Balance(
+                amount=amount,
+                usd_value=amount * usd_price,
+            )
+
+        return dict(assets_balance), ''
+
+    def query_online_trade_history(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> tuple[list[Trade], tuple[Timestamp, Timestamp]]:
+        """
+        https://www.okx.com/docs-v5/en/#rest-api-trade-get-order-history-last-3-months
+        """
+        raw_trades = self._paginated_api_query('trades', 'ordId', {
+            'start_ts': start_ts,
+            'end_ts': end_ts,
+        })
+
+        trades: list[Trade] = []
+        for raw_trade in raw_trades:
+            trade = self.trade_from_okx(raw_trade)
+            if trade is not None:
+                trades.append(trade)
+
+        return trades, (start_ts, end_ts)
+
+    def query_online_margin_history(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[MarginPosition]:
+        return []  # noop for okx
+
+    def query_online_deposits_withdrawals(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[AssetMovement]:
+        """
+        https://www.okx.com/docs-v5/en/#rest-api-funding-get-deposit-history
+        https://www.okx.com/docs-v5/en/#rest-api-funding-get-withdrawal-history
+        """
+        deposits = self._paginated_api_query('deposits', 'ts')
+        withdrawals = self._paginated_api_query('withdrawals', 'ts')
+
+        movements = []
+        for raw_movement in deposits:
+            movement = self.asset_movement_from_okx(
+                raw_movement,
+                AssetMovementCategory.DEPOSIT,
+            )
+            if movement is not None:
+                movements.append(movement)
+        for raw_movement in withdrawals:
+            movement = self.asset_movement_from_okx(
+                raw_movement,
+                AssetMovementCategory.WITHDRAWAL,
+            )
+            if movement is not None:
+                movements.append(movement)
+
+        return movements
+
+    def query_online_income_loss_expense(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[LedgerAction]:
+        return []  # noop for okx
+
+    def trade_from_okx(self, raw_trade: dict[str, Any]) -> Optional[Trade]:
+        """
+        Converts a raw trade from OKX into a Trade object
+        """
+        try:
+            timestamp = ts_ms_to_sec(TimestampMS(int(raw_trade['cTime'])))
+            [base_asset_str, quote_asset_str] = raw_trade['instId'].split('-')
```

Comment:
In python the `[]` is not needed for assignment

```suggestion
            base_asset_str, quote_asset_str = raw_trade['instId'].split('-')
```